### DISCUSSION TITLE
How to limit the number of `textDocument/completion` operation results?

### DISCUSSION THREAD

**Discussion by zhouyusd** [microsoft/pyright#7234] [Q&A] [answered]
Sometimes, the 'textDocument/completion' operation returns too many results, this can cause lag when editing with the monaco editor, and I want to know how to limit the number of returned results. like clangd option `--limit-results=<int>           - Limit the number of results returned by clangd. 0 means no limit (default=100)`

**Comment by erictraut** [comment, marked as answer]
Pyright's completion provider (the component that generates completion suggestions) is relatively simple. It doesn't attempt to limit the number of completion results. The completion provider in Pylance is much more sophisticated and has many optimizations (including indexing) that are not found in pyright's completion provider. You might want to try out Pylance (which runs in VS Code) to see if that meets your requirements.

**Reply by zhouyusd** [reply]
Thanks very much.
